### PR TITLE
Docs Generation

### PR DESCRIPTION
**Pull request by TriplEight**
Generates docs on merges, schedules (if any), tags.
Can add `only: changes: "*.rs"` if needed.

### CODE REVIEW

**Comment by Robbepop** [comment]
> Can add `only: changes: "*.rs"` if needed.

What does this do?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -195,3 +196,28 @@ examples-generate-metadata:
...
+
+#### stage:                        publish
+
+publish-docs:
+  stage:                           publish
+  <<:                              *docker-env
+  only:
+    - master
+    - schedules
+    - tags
```
Comment (Robbepop) [inline .gitlab-ci.yml]: Not sure we need this outside merged to `master` tbh

**Comment by TriplEight** [comment]
> What does this do? 

pipeline will only launch if `*.rs` (or any other type, or a dir if you want) files were changed.

File: .gitlab-ci.yml (hunk 1 above)
Comment (TriplEight) [reply .gitlab-ci.yml]: - `schedules` can be a nightly job on `master`, we just didn't need it before.
- `tags` will do a documentation update when you release.
Don't you need any of those?

**Pull request merged**